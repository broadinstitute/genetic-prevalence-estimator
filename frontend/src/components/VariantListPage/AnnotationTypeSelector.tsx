import {
  FormControl,
  FormLabel,
  HStack,
  Radio,
  RadioGroup,
  Text,
} from "@chakra-ui/react";

export type AnnotationOption = "shared" | "personal";

type AnnotationTypeSelectorProps = {
  userCanEdit: boolean;
  value: AnnotationOption;
  onChange: (value: AnnotationOption) => void;
};

const AnnotationTypeSelector = (props: AnnotationTypeSelectorProps) => {
  const { userCanEdit, value, onChange } = props;

  return (
    <>
      {!userCanEdit && <Text>Viewing shared annotations</Text>}
      {userCanEdit && (
        <FormControl id="annotation-type-selector" as="fieldset">
          <FormLabel as="legend">Annotation type</FormLabel>
          <RadioGroup
            value={value}
            onChange={(value: AnnotationOption) => {
              onChange(value);
            }}
          >
            <HStack spacing="24px">
              <Radio value="shared">Shared</Radio>
              <Radio value="personal">Personal</Radio>
            </HStack>
          </RadioGroup>
        </FormControl>
      )}
    </>
  );
};

export default AnnotationTypeSelector;
