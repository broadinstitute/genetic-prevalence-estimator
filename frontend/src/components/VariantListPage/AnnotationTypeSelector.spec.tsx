import { render } from "@testing-library/react";

import AnnotationTypeSelector from "./AnnotationTypeSelector";

describe("AnnotationTypeSelector", () => {
  test("is a single line of text about viewing shared annotations if the user does not have edit permissions", () => {
    const result = render(
      <AnnotationTypeSelector
        userCanEdit={false}
        value={"shared"}
        onChange={() => {
          return;
        }}
      />
    );
    expect(result.asFragment()).toMatchSnapshot();
  });

  test("is a component that allows selection between shared and personal annotations if the user has edit permissions", () => {
    const result = render(
      <AnnotationTypeSelector
        userCanEdit={true}
        value={"shared"}
        onChange={() => {
          return;
        }}
      />
    );
    expect(result.asFragment()).toMatchSnapshot();
  });
});
