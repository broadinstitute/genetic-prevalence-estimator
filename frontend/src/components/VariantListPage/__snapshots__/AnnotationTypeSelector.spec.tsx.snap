// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnnotationTypeSelector is a component that allows selection between shared and personal annotations if the user has edit permissions 1`] = `
<DocumentFragment>
  <fieldset
    class="chakra-form-control css-1kxonj9"
    role="group"
  >
    <legend
      class="chakra-form__label css-uvho5g"
      for="annotation-type-selector"
      id="annotation-type-selector-label"
    >
      Annotation type
    </legend>
    <div
      class="chakra-radio-group css-0"
      role="radiogroup"
    >
      <div
        class="chakra-stack css-1y5q8ss"
      >
        <label
          class="chakra-radio css-78joka"
        >
          <input
            checked=""
            class="chakra-radio__input"
            id="annotation-type-selector"
            name="radio-1"
            style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
            type="radio"
            value="shared"
          />
          <span
            aria-hidden="true"
            class="chakra-radio__control css-ssalds"
            data-checked=""
          />
          <span
            class="chakra-radio__label css-1y8kf23"
            data-checked=""
          >
            Shared
          </span>
        </label>
        <label
          class="chakra-radio css-78joka"
        >
          <input
            class="chakra-radio__input"
            id="annotation-type-selector"
            name="radio-1"
            style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
            type="radio"
            value="personal"
          />
          <span
            aria-hidden="true"
            class="chakra-radio__control css-ssalds"
          />
          <span
            class="chakra-radio__label css-1y8kf23"
          >
            Personal
          </span>
        </label>
      </div>
    </div>
  </fieldset>
</DocumentFragment>
`;

exports[`AnnotationTypeSelector is a single line of text about viewing shared annotations if the user does not have edit permissions 1`] = `
<DocumentFragment>
  <p
    class="chakra-text css-0"
  >
    Viewing shared annotations
  </p>
</DocumentFragment>
`;
